lib-clay/hashmaps/hashmaps.clay

```
@@ -213,3 +220,28 @@ private findInBucket(_node:NodePtr[K,V], key:K) {
     }
     return null(V);
 }
+
+[K,V]
+private grow(a:HashMap[K,V]) {
+    var oldbuckets = a.buckets;
+
+    a.buckets <-- Vector[NodePtr[K,V]]();
+    a.size = 0;
+
+    try {
+        resize(a.buckets, size(oldbuckets) * 2);
+    } catch (ex) {
+        destroy(a.buckets);
+        throw ex;
+    }
```

So I can just `resize(a.buckets, size(oldbuckets) * 2)` and ignore any exception?